specs/protocol/holocene/configurability.md

```diff
@@ -0,0 +1,89 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Configurability](#configurability)
+  - [Overview](#overview)
+  - [Constants](#constants)
+    - [`ConfigType`](#configtype)
+  - [`SystemConfig`](#systemconfig)
+    - [Interface](#interface)
+  - [`OptimismPortal`](#optimismportal)
+    - [Interface](#interface-1)
+      - [`setConfig`](#setconfig)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Configurability
+
+## Overview
+
+The `SystemConfig` and `OptimismPortal` are updated with a new flow for chain
+configurability. This new flow reduces the need for the `ConfigUpdate` events
+emitted by the `SystemConfig`, which adds great cost to the fault proof program's
+execution as it results in the program to iterate through all possible logs to
+ensure that it has observed all of the events. It also enables the L1 Attributes
+transaction to not post duplicate data between transactions which reduces database
+size growth over time.
+
+## Constants
+
+### `ConfigType`
+
+The `ConfigType` enum represents configuration that can be modified.
+
+| Name | Value | Description |
+| ---- | ----- | --- |
+| `SET_GAS_LIMIT` | `uint8(0)` | Sets the L2 gas limit |
+| `SET_FEE_SCALARS` | `uint8(1)` | Sets the L2 fee configuration |
+| `SET_EIP_1559_PARAMS` | `uint8(2)` | Sets the EIP-1559 elasticity and denominator |
+| `SET_BATCHER_HASH` | `uint8(3)` | Sets the batcher role |
+| `SET_UNSAFE_BLOCK_SIGNER` | `uint8(4)` | Sets the p2p sequencer key |
+| `SET_GAS_PAYING_TOKEN` | `uint8(5)` | Modifies the gas paying token for the chain |
```

Comment:
The `SET_GAS_PAYING_TOKEN` functionality is currently implemented in the Interop contract: https://github.com/ethereum-optimism/optimism/blob/aebf669c108240298e14bd0e29068a18e0b0a52e/packages/contracts-bedrock/src/L2/L1BlockInterop.sol#L63 . Should we consider moving this implementation into the Holocene contract?

Reply:
Yes we will want to move this to the holocene contract